moto/dynamodb2/responses.py

```diff
@@ -826,7 +826,7 @@ def _build_updated_new_attributes(self, original, changed):
                         original.get(key, None), changed[key]
                     )
                     for key in changed.keys()
-                    if changed[key] != original.get(key, None)
+                    if key not in original or changed[key] != original[key]
```

Comment:
here, would we also need to guard against the case where the attribute exists but was originally an explicit `None`, eg, check
```
if original.get(key, None) is not None ...
```